From 6c61cac36b69502916f185fb2c527ad337f3b450 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Mon, 12 Sep 2022 12:10:44 +0300
Subject: [PATCH] [SW-0] gaudi3: remove usage of tpm validation flag

Currently, TPM implementation in ROM extends ppboot PCR based on
tpm validation flag which is cleared on cold reset and fw update.
Due to new requirement to clear PCR memory before extending PCR, TPM
needs to be reset via external reset control and needs to be reset on
every reset. This patch removes check of tpm validation flag in TPM ROM
implementation which means TPM will extend ppboot PCR unconditionally.
ROM will still update the indication of tpm validation in cold flops
for debug purpose.

ROM assumes TPM reset is perform by CPLD before any reset.

Change-Id: Ie5cf75a15c11272051d70eb42c9d5333c1db4b41
---

diff --git a/app/arcboot_rom.c b/app/arcboot_rom.c
index 050c3d8..bade748 100644
--- a/app/arcboot_rom.c
+++ b/app/arcboot_rom.c
@@ -27,7 +27,6 @@
 	.sku = 0,
 	.fuse_spi_offset = 0x0,
 	.fuse_spi = false,
-	.tpm_validation_pass = false,
 };
 
 #define ARCBOOT_STATUS_REG_SZ_BITS		32
diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index 967b67c..4890352 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -412,7 +412,6 @@
 	arcboot_rom_global.recovery_flag = cold_rst_data.recovery_flag;
 	arcboot_rom_global.validation_pass = cold_rst_data.validation_flag;
 	arcboot_rom_global.efuse_read = cold_rst_data.efuse_read_flag;
-	arcboot_rom_global.tpm_validation_pass = cold_rst_data.tpm_validation_flag;
 }
 
 static void arcboot_rom_persist_data_non_zero_init(void)
@@ -909,11 +908,6 @@
 		   arcboot_rom_status_update(PROGRESS_ARCBOOT_TPM_SPI_FAIL);
 	} else {
 		arcboot_rom_status_update(PROGRESS_ARCBOOT_ROM_TPM);
-
-		if (arcboot_rom_global.tpm_validation_pass) {
-			arcboot_rom_status_update(PROGRESS_ARCBOOT_ROM_TPM_FINISH);
-			return;
-		}
 
 		if (!arcboot_rom_tpm_rom()) {
 			arcboot_rom_tpm_validation(true);
diff --git a/include/arcboot.h b/include/arcboot.h
index 0710c83..76504c8 100644
--- a/include/arcboot.h
+++ b/include/arcboot.h
@@ -101,7 +101,6 @@
 	bool fuse_spi;
 	struct hl_fw_boot_ptrs_ctx boot_ptr_ctx[HL_FW_BOOT_PTR_MAX_IMG];
 	struct hl_fw_sec_info sec_data;
-	uint8_t tpm_validation_pass;
 	struct hl_fw_img_desc spi_img_desc[ROM_DESC_NUM_MAX];
 	struct hl_fw_boot_ptrs spi_manifest_hdr;
 };
